### PR TITLE
Add spec for ClzOp

### LINKED ISSUE #448: Add spec for ClzOp

### PR DESCRIPTION
**Pull request by ghpvnist**
closes #448

### CODE REVIEW

**Review by subhankarshah — APPROVED** [review]
LGTM

File: docs/spec_draft.md [hunk 1]
```diff
@@ -761,6 +762,39 @@ specification. Numeric precision is implementation-defined.
...
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same type.
+
+### Examples
+
+```mlir
+// %operand: [[0, 1], [127, -1]]
+%result = "stablehlo.count_leading_zeros"(%operand) : (2x2xi8) -> tensor<2x2xi8>
```
Comment (burmako) [inline docs/spec_draft.md:792]: This is not a valid MLIR syntax. Fixed: `%result = "stablehlo.count_leading_zeros"(%operand) : (tensor<2x2xi8>) -> tensor<2x2xi8>`.

File: stablehlo/dialect/StablehloOps.td [hunk 2]
```diff
@@ -271,15 +271,15 @@ def StableHLO_ClzOp: StableHLO_UnaryElementwiseOp<"count_leading_zeros",
...
-    See
-    https://www.tensorflow.org/xla/operation_semantics#element-wise_unary_functions.
+    See:
+    https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#stablehlocount_leading_zeros
 
     Example:
-
     ```mlir
-    %0 = stablehlo.count_leading_zeros %arg0 : tensor<2x2xi32>
+    %result = stablehlo.count_leading_zeros %operand : tensor<2x2xi32>
```
Comment (burmako) [inline stablehlo/dialect/StablehloOps.td:282]: The example above is using `tensor<2x2xi8>`.

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec_draft.md (hunk 1 above)
Comment (ghpvnist) [reply docs/spec_draft.md]: Thanks for catching this!

**Review by burmako — APPROVED** [review]

**Pull request merged**